mmf/models/vilbert.py

```diff
@@ -1125,10 +1125,16 @@ def __init__(self, config):
         classifier_config.hidden_size = config.bi_hidden_size
         if self.config.training_head_type == "nlvr2":
             classifier_config.hidden_size *= 2
-        self.classifier = nn.Sequential(
-            BertPredictionHeadTransform(classifier_config),
-            nn.Linear(classifier_config.hidden_size, self.num_labels),
-        )
+
+        if self.config.training_head_type == 'visual7w':
```

Comment:
You don't need a separate head, you can use classification one directly with num_labels defined.

Reply:
Keeping the head due to difference in head with BertPredictionHeadTransform and just a linear layer. We are addressing this in Flickr30K diff.